test/basictests.jl

```diff
@@ -570,29 +570,38 @@ end
     # These are tests for NewtonRaphson so we should set alpha_initial to be high so that we
     # converge quickly
 
-    function benchmark_nlsolve_oop(f, u0, p = 2.0; alpha_initial = 10.0)
+    function benchmark_nlsolve_oop(f,
+            u0,
+            p = 2.0;
+            alpha_initial = 10.0,
+            update_alpha = switched_evolution_relaxation)
         prob = NonlinearProblem{false}(f, u0, p)
-        return solve(prob, PseudoTransient(; alpha_initial), abstol = 1e-9)
+        return solve(prob, PseudoTransient(; alpha_initial, update_alpha), abstol = 1e-9)
     end
 
     function benchmark_nlsolve_iip(f, u0, p = 2.0; linsolve, precs,
-            alpha_initial = 10.0)
+            alpha_initial = 10.0, update_alpha = switched_evolution_relaxation)
         prob = NonlinearProblem{true}(f, u0, p)
-        return solve(prob, PseudoTransient(; linsolve, precs, alpha_initial), abstol = 1e-9)
+        return solve(prob,
+            PseudoTransient(; linsolve, precs, alpha_initial, update_alpha),
+            abstol = 1e-9)
     end
 
-    @testset "PT: alpha_initial = 10.0 PT AD: $(ad)" for ad in (AutoFiniteDiff(),
-        AutoZygote())
+    @testset "update_alpha:$(update_alpha) alpha_initial = $(alpha_initial) AD: $(ad)" for (update_alpha, alpha_initial) in zip((switched_evolution_relaxation,
+                robust_update_alpha),
+            (10.0, 0.01)),
```

Comment:
Why are you having to change the alpha for a "robust" method? A robust method is by definition supposed to work without having to tune alpha

Reply:
We still have to choose an initial alpha for these problems. The problems were designed to test NewtonRaphson, hence it helps to set the initial alpha slightly higher. There is no single initial alpha that works for every problem. I had to make a guess that works for these problems. The point here is that robust one converges even if we start with a much lower alpha compared to the regular PT with the default maxiters. I can set this to small alpha, but it would just take more iterations. From our discussion, what I understood was Robust one will be slower, but it will try to find a stable solution.

Reply:
So this works if the alpha is not changed? In which case it should not be changed

Reply:
Also you are testing for a problem with an unstable equilibrium, either change it to

1. Problem with a stable equilibrium
2. initialize correctly so that the vector field doesn't make the euler diverge